subsys/bluetooth/mesh/main.c

```diff
@@ -118,32 +115,29 @@ int bt_mesh_provision(const uint8_t net_key[16], uint16_t net_idx,
 		err = bt_mesh_cdb_node_key_import(node, dev_key);
 		if (err) {
 			LOG_ERR("Failed to import cdb device key");
-			goto end;
-		}
-
-		if (IS_ENABLED(CONFIG_BT_SETTINGS)) {
-			bt_mesh_cdb_node_store(node);
+			goto error_exit;
 		}
+		bt_mesh_cdb_node_store(node);
```

Comment:
This needs to be aborted if keys importing failed (or network creation failed, or if anything else in bt_mesh_provision failed).

Reply:
Not needed since this is deallocated below.

Reply:
It will try to store and erase. Or just erase. In either case it will do write to flash which is not needed.